pkg/retrieval/retrieval.go

```diff
@@ -181,12 +179,14 @@ func (s *Service) retrieveChunk(ctx context.Context, addr swarm.Address, skipPee
 		return nil, peer, fmt.Errorf("read delivery: %w peer %s", err, peer.String())
 	}
 
-	// credit the peer after successful delivery
 	chunk = swarm.NewChunk(addr, d.Data)
-	if !s.validator.Validate(chunk) {
-		return nil, peer, err
+	if !content.Valid(chunk) {
+		if !soc.Valid(chunk) {
```

Comment:
because then you must execute both validators to resolve the `if`.
eg:
`if !content.Valid || !soc.Valid` means you will always execute both (since always one validator should be true).
using `&&` would have the same result.
the assumption is that most chunks are content addressed, and so we'd like to assume most will pass the first validator, and that as of such we don't want to run second validator